src/pages/api/events/[id].ts

```diff
@@ -0,0 +1,46 @@
+import { type NextApiRequest, type NextApiResponse } from "next";
+import { request } from "graphql-request";
+import { GET_EVENT_BY_ID } from "~/queries/get-event-by-id";
+import { type Result } from "./all";
+import convertDate from "~/utils/date-converter";
+const endpoint = process.env.EVENTS_ENDPOINT;
+
+interface Event {
+  event: Result | null;
+}
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  const query = req.query;
+  const variables = {
+    eventId: query.id,
+  };
+  try {
+    const { event }: Event = await request(
+      endpoint,
+      GET_EVENT_BY_ID,
+      variables,
+      {
+        Authorization: process.env.EVENTS_KEY,
+      }
+    );
+    let result;
```

Comment:
How about the code below instead of using `let result`?

```suggestion
    if (event) {
    const  result = {
       id: event.id,
       title: event.title,
       eventUrl: event.eventUrl,
       description: event.description,
       venue: event.venue,
       imageUrl: event.image.baseUrl,
       dateTime: convertDate(event.dateTime),
  };
    res.status(200).json(result);
}
```

Reply:
I'm just wondering how you'd deal with the case event is `null` with the code you provided above.
Will you add `res.status(404).json({result: null})` or something?

Reply:
@Lada496 Yeah, that's the problem... I was just wondering if event is falsy, that would result in returing `res.status(400)`, but if this idea is wrong, plz ignore my comment here (Probably I gotta look into exactly how try catch statement works...)

Reply:
Meetup API returns `null` without any error when we try to get `event` by `id`. So we can't catch it as an error.
I'll just change update code to explicitly tell event is `null`.